packages/typespec-python/test/mock_api_tests/asynctests/test_payload_content_negotiation_async.py

```diff
@@ -0,0 +1,53 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import Iterator
+import base64
+import pytest
+from pathlib import Path
+from payload.contentnegotiation.aio import ContentNegotiationClient
+from payload.contentnegotiation.models import PngImageAsJson
+
+
+FILE_FOLDER = Path(__file__).parent.parent
+
+@pytest.fixture
+async def client():
+    async with ContentNegotiationClient(endpoint="http://localhost:3000") as client:
+        yield client
+
+@pytest.fixture
+def png_data() -> bytes:
+    with open(str(FILE_FOLDER / "data/image.png"), "rb") as file_in:
+        return file_in.read()
+
+@pytest.fixture
+def jpg_data() -> bytes:
+    with open(str(FILE_FOLDER / "data/image.jpg"), "rb") as file_in:
+        return file_in.read()
+
+async def iter_bytes_to_bytes(data: Iterator[bytes]) -> bytes:
+    return b"".join([chunk async for chunk in data])
+
+@pytest.mark.asyncio
+async def test_get_avatar_as_png(client: ContentNegotiationClient, png_data: bytes):
+    result = await client.same_body.get_avatar_as_png(stream=True)
```

Comment:
cadl-ranch [return binary data](https://github.com/Azure/cadl-ranch/blob/c7e83c056235000a5942b2aac8c4ba74852816ae/packages/cadl-ranch-specs/http/payload/content-negotiation/mockapi.ts#L22-L23). If I don't `set stream=True`, the following error happens in azure-core:
![image](https://github.com/Azure/autorest.python/assets/70930885/f26cff4b-1052-40a7-a5d9-27940af33e03)